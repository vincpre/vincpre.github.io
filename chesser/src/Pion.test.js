/* eslint-env jest */
const Pion = require("./Pion");

describe("Pion", () => {
  describe("1er dÃ©placement", () => {
    it("en Y de 1 case", () => {
      const pion = new Pion("Pion", 1, 2, "blanc");
      expect(pion.joue(1, 3)).toMatchSnapshot();
    });
    it("en Y de 2 cases", () => {
      const pion = new Pion("Pion", 1, 2, "blanc");
      expect(pion.joue(1, 4)).toMatchSnapshot();
    });
    it("en Y de 3 cases", () => {
      const pion = new Pion("Pion", 1, 2, "blanc");
      expect(pion.joue(1, 5)).toMatchSnapshot();
    });
  });
});
