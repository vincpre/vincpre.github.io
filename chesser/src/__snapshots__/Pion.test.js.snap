// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pion 1er déplacement en Y de 1 case 1`] = `
Pion {
  "coordonnees": Point {
    "x": 1,
    "y": 3,
  },
  "couleur": "blanc",
  "deplacements": Array [
    Point {
      "x": 0,
      "y": 1,
    },
  ],
  "nom": "Pion",
  "nombreDeCoupJoue": 1,
  "prises": Array [
    Point {
      "x": 1,
      "y": 1,
    },
    Point {
      "x": 1,
      "y": 1,
    },
  ],
}
`;

exports[`Pion 1er déplacement en Y de 2 cases 1`] = `
Pion {
  "coordonnees": Point {
    "x": 1,
    "y": 4,
  },
  "couleur": "blanc",
  "deplacements": Array [
    Point {
      "x": 0,
      "y": 1,
    },
  ],
  "nom": "Pion",
  "nombreDeCoupJoue": 1,
  "prises": Array [
    Point {
      "x": 1,
      "y": 1,
    },
    Point {
      "x": 1,
      "y": 1,
    },
  ],
}
`;

exports[`Pion 1er déplacement en Y de 3 cases 1`] = `
Pion {
  "coordonnees": Point {
    "x": 1,
    "y": 2,
  },
  "couleur": "blanc",
  "deplacements": Array [
    Point {
      "x": 0,
      "y": 1,
    },
    Point {
      "x": 0,
      "y": 2,
    },
  ],
  "nom": "Pion",
  "nombreDeCoupJoue": 0,
  "prises": Array [
    Point {
      "x": 1,
      "y": 1,
    },
    Point {
      "x": 1,
      "y": 1,
    },
  ],
}
`;
